### PR TITLE
[Docs] Add installer guide

### PR DESCRIPTION
**Pull request by hydai**
Replace #2207

### CODE REVIEW

File: docs/book/en/src/contribute/installer.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+* Short Option: `-h`
+* Full Option: `--help`
+* Description: Show this help message and exit.
+
+### Verbose
+* Short Option: `-D`
+* Full Option: `--debug`
+* Description: Enable verbosity debug
+
+### Specify Installed Version
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:55]: Specify the version of WasmEdge to install

File: docs/book/en/src/contribute/installer.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+* Description: Install the given VERSION of WasmEdge
+* Available Value: VERSION `0.11.2` or other valid release versions.
+* Note - In case if an invalid or non existing version is suppliead the installer exits with an error.
+
+### Install Path
+* Short Option: `-p PATH`
+* Full Option: `--path PATH`
+* Description: Install WasmEdge into the given PATH. The default Path is `$HOME/.wasmedge`.
+* Note - In any path other than the ones starting with `/usr` are treated as non system paths in the internals of the installer. The consequences are different directory structures for both.
+* Note - If the path does not exist it is created.
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:67]: If the path not exists, the folder will be created.

File: docs/book/en/src/contribute/installer.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+* Short Option: `-D`
+* Full Option: `--debug`
+* Description: Enable verbosity debug
+
+### Specify Installed Version
+* Short Option: `-v VERSION`
+* Full Option: `--version VERSION`
+* Description: Install the given VERSION of WasmEdge
+* Available Value: VERSION `0.11.2` or other valid release versions.
+* Note - In case if an invalid or non existing version is suppliead the installer exits with an error.
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:60]: In the case of supplied an invalid or nonexistent version, the installer exists with an error.

File: docs/book/en/src/contribute/installer.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+### Install Path
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:62]: Installation path

File: docs/book/en/src/contribute/installer.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+* Short Option: `-p PATH`
+* Full Option: `--path PATH`
+* Description: Install WasmEdge into the given PATH. The default Path is `$HOME/.wasmedge`.
+* Note - In any path other than the ones starting with `/usr` are treated as non system paths in the internals of the installer. The consequences are different directory structures for both.
+* Note - If the path does not exist it is created.
+
+### Uninstall existed version
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:69]: Uninstallation

File: docs/book/en/src/contribute/installer.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+### Install Extensions
+* Short Option: `-e [EXTENSIONS [EXTENSIONS ...]]`
+* Full Option: `--extension [EXTENSIONS [EXTENSIONS ...]]`
+* Description: Install wasmedge-extension tools.
+* Available Value (case sensitive): Supported Extensions `'tensorflow', 'image', 'all'`.
+
+#### Tensorflow Extensions Library Version
+* Full Option: `--tf-version TF_VERSION`
+* Description: Install the given VERSION of the library of the Tensorflow and Tensorflow lite extension. Only available when the `Extensions` is set to `all` or `tensorflow`.
+* Note - It's value is same as that of wasmedge version if not specified.
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:90]: the same as the WasmEdge version

File: docs/book/en/src/contribute/installer.md [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Tensorflow Extensions Dependencies Version
+* Full Option: `--tf-deps-version TF_DEPS_VERSION`
+* Description: Install the given VERSION of the dependencies of the Tensorflow and Tensorflow lite extension. Only available when the `Extensions` is set to `all` or `tensorflow`.
+* Note - It's value is same as that of wasmedge version if not specified.
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:95]: the same as the WasmEdge version

File: docs/book/en/src/contribute/installer.md [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Tensorflow Extensions Tools Version
+* Full Option: `--tf-tools-version TF_TOOLS_VERSION`
+* Description: Install the given VERSION of the tools of the Tensorflow and Tensorflow lite extension. Only available when the `Extensions` is set to `all` or `tensorflow`.
+* Note - It's value is same as that of wasmedge version if not specified.
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:100]: the same as the WasmEdge version

File: docs/book/en/src/contribute/installer.md [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Image Extensions Version
+* Full Option: `--image-version IMAGE_VERSION`
+* Description: Install the given VERSION of the Image extension. Only available when the `Extensions` is set to `all` or `image`.
+* Note - It's value is same as that of wasmedge version if not specified.
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:105]: the same as the WasmEdge version

File: docs/book/en/src/contribute/installer.md [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+#### Image Extensions Dependencies Version
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:107]: Remove this column.
image-deps has been removed from installer due to deprecating 0.8.2.

File: docs/book/en/src/contribute/installer.md [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+### Plugins
+
+* Note - Currently `--plugins` is an experimental option.
+
+* Full Option: `--plugins wasi_crypto:0.11.0`
+
+* Note - The format for this argument is `<plugin_name>:<version_number>`. `<version_number>` is not compulsory. For example `--plugins wasi_crypto` is a valid option.
+* Note - `<plugin_name>` is cases sensitive. Allowed values are stated [here](https://wasmedge.org/book/en/plugin.html) in the `Rust Crate` column. The logic is that the release name should be the same.
+* Note - It's value is same as that of wasmedge version if not specified.
```
Comment (q82419) [inline docs/book/en/src/contribute/installer.md:120]: the same as the WasmEdge version

**Review by q82419 — CHANGES_REQUESTED** [review]

**Comment by hydai** [comment]
All comments are fixed.

**Review by q82419 — APPROVED** [review]

**Pull request merged**